For issue #12126 - Tab counter consumeFrom update

BMEduard opened the pull request:
Modified the consumeFrom method inside HomeFragment for the tab counter

### Pull Request checklist
<!-- Before submitting the PR, please address each item -->
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

ekager commented:
Now I'm remembering the reason I switched this in the first place was because I couldn't get UI tests to run and I didn't know why we were using it anyway :/ Looks like the UI tests are crashing on
`kotlin.TypeCastException: null cannot be cast to non-null type org.mozilla.fenix.HomeActivity
at org.mozilla.fenix.home.HomeFragment.getBrowsingModeManager(HomeFragment.kt:131)
at org.mozilla.fenix.home.HomeFragment.access$getBrowsingModeManager$p(HomeFragment.kt:115)
at org.mozilla.fenix.home.HomeFragment$onViewCreated$9.invoke(HomeFragment.kt:393)
at org.mozilla.fenix.home.HomeFragment$onViewCreated$9.invoke(HomeFragment.kt:115)
at mozilla.components.lib.state.ext.ViewKt$consumeFrom$1.invokeSuspend(View.kt:37)
at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)`

BMEduard commented:
> Now I'm remembering the reason I switched this in the first place was because I couldn't get UI tests to run and I didn't know why we were using it anyway :/ Looks like the UI tests are crashing on ...

After investigating this issue I found that the UI tests were crashing because the consumeFrom ext function from View doesn't account for the case where the fragment isn't added to the activity, causing the call to the activity's BrowsingModeManager to fail. The ext function from Fragment does account for this and checks if the fragment is added to it's activity.
However, the ext function from Fragment does not invoke the updateTabCounter block in the scenarios described in #12126, so I added an direct call to the method to account for this.
